src/main/scala/com/campudus/tableaux/database/model/CellAnnotationConfigModel.scala

```diff
@@ -0,0 +1,199 @@
+    }
+  }
+
+  def delete(name: String): Future[Unit] = {
+    val delete = s"DELETE FROM $table WHERE name = ?"
+
+    for {
+      result <- connection.query(delete, Json.arr(name))
+      _ <- Future(deleteNotNull(result))
+    } yield ()
+  }
+
+  def create(
+      name: String,
+      priority: Option[Int],
+      fgColor: String,
+      bgColor: String,
+      displayName: MultiLanguageValue[String],
+      isMultilang: Option[Boolean],
+      isDashboard: Option[Boolean]
+  )(implicit user: TableauxUser): Future[String] = {
+
+    val insert = s"""INSERT INTO $table (
+                    |  name,
+                    |  priority,
+                    |  fg_color,
+                    |  bg_color,
+                    |  display_name,
+                    |  is_multilang,
+                    |  is_dashboard)
+                    |VALUES
+                    |  (?, ?, ?, ?, ?, ?, ?) RETURNING name""".stripMargin
+
+    for {
+      _ <- checkUniqueName(name)
+      newPriority <- getNewPriority()
+      result <- connection.query(
+        insert,
+        Json
+          .arr(
+            name,
+            priority.getOrElse(newPriority),
+            fgColor,
+            bgColor,
+            displayName.getJson.toString,
+            isMultilang.getOrElse(false),
+            isDashboard.getOrElse(true)
+          )
+      )
+
+      configName = insertNotNull(result).head.get[String](0)
+    } yield configName
+
+  }
+
+  def retrieveAll()(implicit user: TableauxUser): Future[Seq[CellAnnotationConfig]] = {
+    for {
+      result <- connection.query(selectStatement(None))
+      resultArr <- Future(resultObjectToJsonArray(result))
+    } yield {
+      resultArr.map(convertJsonArrayToCellAnnotationConfig)
+    }
+  }
+
+  private def convertJsonArrayToCellAnnotationConfig(arr: JsonArray)(implicit
+  user: TableauxUser): CellAnnotationConfig = {
+    CellAnnotationConfig(
+      arr.get[String](0), // name
+      arr.get[Int](1), // priority
+      arr.get[String](2), // fgColor
+      arr.get[String](3), // bgColor
+      MultiLanguageValue.fromString(arr.get[String](4)), // displayName
+      arr.get[Boolean](5), // isMultilang
+      arr.get[Boolean](6), // isDashboard
+      arr.get[Boolean](7) // isCustom
+    )
+  }
+
+  private def checkUniqueName(name: String): Future[Unit] = {
+    val sql = s"SELECT COUNT(*) = 0 FROM $table WHERE name = ?"
+
+    connection
+      .selectSingleValue[Boolean](sql, Json.arr(name))
+      .flatMap({
+        case true => Future.successful(())
+        case false => Future.failed(ShouldBeUniqueException(
+            s"Name of annotation config should be unique '$name'.",
+            "cellAnnotationConfig"
+          ))
+      })
+  }
+
+  private def getNewPriority(): Future[Int] = {
+    val sql = s"SELECT MAX(priority) FROM $table"
+
+    for {
+      maxPriority <- connection.selectSingleValue[Int](sql)
+    } yield {
+      maxPriority + 1
+    }
+  }
+}
```

Comment:
Du hättest diese "Arbeit" auch direkt von SQL erledigen lassen können, indem du dieses select stmt in den insert mit rein packst. Sind zwar jetzt zwei requests aber diese Schreibvorgänge sind ja nicht performance-kritisch.
So geht es jedenfalls auch, wollts nur gesagt haben

Reply:
Hab das noch angepasst und in den Insert ein `COALESCE(?, (SELECT MAX(priority) FROM $table), 0) + 1` eingefügt.